File: android/src/main/kotlin/com/example/flutterusb/Response.kt
package com.example.flutterusb

//result may contains error
class Response(val result: String,val dataSendLength: Int, val dataReceived: ByteArray){

    override fun toString(): String {
        return super.toString()
    }

    /*

      string toString() {
          std::stringstream ss;
          ss << "{\"result\":" << result << "\"data_send_length\":\"" << data_send_length << "\",\"byte_list\":\"[";
          for (int i(0); i < sizeof(byte_list) - 1; ++i) {
              ss << (int)byte_list[i] << ",";
          }
          ss << (int)byte_list[sizeof(byte_list)-1] << "]\"}";
          return ss.str();
      }
     */

}

File: windows/response.cpp
#include "response.h"

#include <sstream>

Response::Response(string new_result, int new_data_send_length, std::vector<uint8_t> new_byte_list)
{
    data_send_length = new_data_send_length;
    byte_list = new_byte_list;
    result = new_result;
}